File: max.rb
=begin
「アルゴリズムをはじめよう」第３章　５
配列の最大値を返す
=end

module My

	def max(array)

		max = 0

		array.each do |n|
			 max = n if max < n
		end

		max

	end
	module_function :max

end

p My.max([2,3,5,7,9]) === 9
p My.max([1,1,0,-4,1]) === 1
p My.max([43,7,22,100,0,83]) === 100

File: max.php
<?php

/**
 * 「アルゴリズムをはじめよう」３－５
 * 配列の最大値を返す
 */
function my_max($array){

	$max = 0;

	foreach($array as $num){
		if($max < $num) $max = $num;
	}

	return $max;
}


echo my_max(array(2,3,5,7,9)) === 9;
echo my_max(array(1,1,0,-4,1)) === 1;
echo my_max(array(43,7,22,100,0,83)) === 100;

